solana/src/main/java/org/p2p/solanaj/ws/SocketClientCreateResult.kt

```diff
@@ -0,0 +1,6 @@
+package org.p2p.solanaj.ws
+
+sealed interface SocketClientCreateResult {
```

Comment:
For me it's not obvious to have wrapped instance.
Why did you recognize to use it in this way ?

Reply:
That's a whole subject to discuss
1) Using a monad wrapper like functional `Either` or `Result` or `Optional` is a better way of doing things instead of implicitly throwing exceptions inside
2) There is a guarantee for you, as a user that this method won't throw anything unexpected and you can remove unnecessary try-catch inside your code
3) This result is explicit - you see that this method can return an instance, or error, and you are forced by the complier to handle this type of thing. sealed result type provides Types and Compile Time Safety

https://elizarov.medium.com/kotlin-and-exceptions-8062f589d07